### PR TITLE
Fixed crash if you save a bundle outside the default bundle folder (#4974)

### PR DESCRIPTION
**Pull request by AMZN-stankowi**
Cherry picked from development

Signed-off-by: stankowi <4838196+AMZN-stankowi@users.noreply.github.com>

### CODE REVIEW

**Comment by amzn-pratikpa** [comment]
Any update on this?

**Comment by AMZN-stankowi** [comment]
> 
> 
> Any update on this?

it's a crash fix that should go into stabilization, I'm just waiting on reviews.

File: Code/Tools/AssetBundler/source/models/AssetBundlerAbstractFileTableModel.cpp [hunk 1]
```diff
@@ -61,8 +61,12 @@ namespace AssetBundler
...
             AZStd::string absolutePath = filePath.toUtf8().data();
             if (AZ::IO::FileIOBase::GetInstance()->Exists(absolutePath.c_str()))
-            { 
-                AZStd::string projectName = pathToProjectNameMap.at(absolutePath);
+            {
+                AZStd::string projectName;
+                if (pathToProjectNameMap.contains(absolutePath))
+                {
+                    projectName = pathToProjectNameMap.at(absolutePath);
+                }
```
Comment (lumberyard-employee-dm) [inline Code/Tools/AssetBundler/source/models/AssetBundlerAbstractFileTableModel.cpp:69]: What is the project name in this instance?
Is the O3DE Project name?

If so, why is that needed. There is only one project that can be active in a given application.
At that point AZ::Utils::GetProjectName can just be called.

**Review by jackalbe — APPROVED** [review]

**Review by amzn-mike — APPROVED** [review]

File: Code/Tools/AssetBundler/source/models/AssetBundlerAbstractFileTableModel.cpp (hunk 1 above)
Comment (AMZN-stankowi) [reply Code/Tools/AssetBundler/source/models/AssetBundlerAbstractFileTableModel.cpp:69]: It's not always the O3DE Project Name, it's the name of the thing that holds the object at that path: Gem name, scan folder, or game project. It's not a great name for that container, it could be better named.

For example, the key "C:\\git\\o3de\\Gems\\Atom\\Bootstrap\\Assets\\seedList.seed" resolves to "Atom_Bootstrap Gem", and the key "C:\\git\\o3de\\Assets\\Engine\\SeedAssetList.seed", resolves to "Assets\\Engine"

Comment (lumberyard-employee-dm) [reply Code/Tools/AssetBundler/source/models/AssetBundlerAbstractFileTableModel.cpp:69]: Thanks for the clarification.
So it is really the key supplied for the ScanFolder?

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Tools/AssetBundler/source/models/AssetBundlerAbstractFileTableModel.cpp (hunk 1 above)
Comment (AMZN-stankowi) [reply Code/Tools/AssetBundler/source/models/AssetBundlerAbstractFileTableModel.cpp:69]: Mostly, yeah. It's a human readable display name for the scan folder, but that's not a concept for the scan folder directly, at least in the asset database, so I think the bundler generates that in other ways.
